packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -1033,30 +1024,15 @@ export const Table = <
                     });
 
                     return (
-                      <Box
+                      <ColumnHeader<T>
                         {...columnProps}
                         {...column.getDragAndDropProps()}
                         key={columnProps.key}
-                        title={undefined}
-                        ref={(el) => {
-                          columnResizeRef(el, column);
-                        }}
-                        onMouseDown={() => {
-                          isHeaderDirectClick.current = true;
-                        }}
-                        onClick={(e) => {
-                          // Prevents from triggering sort when resizing and mouse is released in the middle of header
-                          if (isHeaderDirectClick.current) {
-                            onClick?.(e);
-                            isHeaderDirectClick.current = false;
-                          }
-                        }}
-                        tabIndex={showSortButton(column) ? 0 : undefined}
-                        onKeyDown={(e) => {
-                          if (e.key == 'Enter' && showSortButton(column)) {
-                            column.toggleSortBy();
-                          }
-                        }}
+                        onClick={onClick}
+                        columnRefs={columnRefs}
+                        column={column}
+                        isHeaderDirectClick={isHeaderDirectClick}
+                        showSortButton={showSortButton}
                       >
                         <>
```

Comment:
All this could below can also probably be moved into `ColumnHeader`. You can do that in a follow-up PR or in the same PR.

Reply:
Moved the `ColumnHeader`s children inside of the component in this [commit](https://github.com/iTwin/iTwinUI/pull/2178/commits/e153e92bdcd2cdcff79721d638f2d71f77875f10).